config/functions.cfg

```diff
@@ -161,6 +163,34 @@ function go_lang {
 
 }
 
+function update_go_version_from_config {
+
+  GO_VERSION_CONFIG_PATH=$GOPATH/src/github.com/multiversx/mx-chain-$ENVIRONMENT-config/goVersion
+  if ! [ -f $GO_VERSION_CONFIG_PATH ]; then
+    echo -e
+    echo -e "No goVersion file configured, using default one:${CYAN}${GO_LATEST_TESTED}${NC}"
+    echo -e
+
+    return
+  fi
+
+  #Get mx-chain-go go version from the mx-chain-$ENVIRONMENT-config repo
+  GO_VERSION=$(cat $GO_VERSION_CONFIG_PATH)
+  if ! [ -z "${GO_VERSION}" ]; then
+        GO_LATEST_TESTED=${GO_VERSION}
+        echo -e
+        echo -e "Updated go version from config, new version is:${CYAN}${GO_LATEST_TESTED}${NC}"
+        echo -e
+
+        return
+  fi
+
+  echo -e
+  echo -e "goVersion file configured, but is empty. Using default version:${CYAN}${GO_LATEST_TESTED}${NC}"
```

Comment:
I think we should error if the go version file does not exist because that is a wrong setup: it will try to use an old default which might not work. I would also remove the possibility to override the `GO_LATEST_TESTED` variable in the variable.cfg file (remove that definition along with the comments)

Reply:
updated the log only, we'll deprecate it once all chains switch to this config